### ISSUE TITLE
Question/Request: Modify capacitor.config.json

### ISSUE THREAD

**Issue by chvonrohr** [ionic-team/capacitor-configure#61]
Besides the Angular environment files, some plugins require configuration in capacitor.config.json

- e.g. capacitor-mixpanel - we need to set mixpanel project (but this is depending on build environment dev/stg/prod)
- e.g. urbanairship-cordova - we set config in `cordova.capacitor.xyz` section - but also here those are environment specific

So we are able to use cap-config (with xml) for adjusting icons, entitlements, versions and display names - but still having some bash scripts to adjust the above ones.

Is there away (or is it planned) to amend capacitor.config.json from cap-config XML configuration?

**Comment by mlynch** [comment]
Yep that is very reasonable. I was planning on adding support for modifying arbitrary json so this would be very doable.

Thinking something like

```yaml
json:
  - file: capacitor.config.json
    set:
      ios:
        scheme: "my-app-scheme://"
```

Would update the target key/value pair or insert the new value. Thoughts?

**Comment by chvonrohr** [comment]
Usually, we have dev settings as default in the repo, so override would be awesome. But adding would be already a huge thing 🙏

**Comment by chvonrohr** [comment]
I saw it was released in 2.0.9 - many thanks.
From my tests, I saw that other childs are removed (e.g. as in your example above, all other XML childs of `ios` than `scheme` are removed...
Is this intended?

**Comment by mlynch** [comment]
@chvonrohr try using `merge` instead of `set`. `set` will remove any other children because it's like assigning a new value to that key. `merge` will instead merge the two trees
